### ISSUE TITLE
libc++/libstdc++ support

### ISSUE THREAD

**Issue by void2012** [zufuliu/llvm-utils#16]
Hi! This project is AMAZING, thank you! 
One question though.
I use LLVM v141_xp build tools to target Windows XP.
However, I want to use the latest C++ library features. Is there is a way to use libc++ or libstdc++ with posix thread model v141_xp on Visual Studio 2022? Thanks!

**Comment by zufuliu** [comment]
On Windows, libc++ is only supported by llvm-mingw, you can download latest version from https://github.com/mstorsjo/llvm-mingw/releases. libstdc++ is only supported by mingw GCC and Clang, there are many builds on https://www.mingw-w64.org/downloads/

**Comment by void2012** [comment]
Yes I know there is the original POSIX version which works on windows, but I want to use the MSVC-compatible driver (Clang-CL) along with Visual Studio IDE. 

libc++ is designed to be compatible with Clang-CL too, I saw some code related to that in libc++ sources. But I would like to know if there is a way to combine that with the toolset to be used by the Visual Studio IDE. 

There is a libc++ as a Visual Studio project though, https://github.com/enlyze/libcxx/

But it have some serious issues, like it fails to recognize things like `__bf16` in intrin.h file and some other builtins even on the newest version of clang and with `/arch:SSE2` flag. It also says symbol `__building_module` (which is a compiler builtin) is undefined no matter you use modules or not. If you try to pass preprocessor it D__building_module(x)=0 as a workaround, it says you can't override a compiler builtin with macros. And so on and so forth...

VC++ directory page is buggy, too. If you try to exclude VC_IncludePath and change the include directory to the include directory of libc++ it still uses MSVC STL.

Sadly I have no Idea how Visual Studio works but my guess is that this can be solved by creating a specific toolset that ships with libc++ as a standard library and maybe MinGW's Windows SDK.
